docs/apidiffs/current_vs_latest/opentelemetry-semconv.txt

```diff
@@ -0,0 +1,170 @@
++++  NEW CLASS: PUBLIC(+) FINAL(+) io.opentelemetry.semconv.AttributeKeyTemplate  (not serializable)
+	+++  CLASS FILE FORMAT VERSION: 52.0 <- n.a.
+	GENERIC TEMPLATES: +++ T:java.lang.Object
+	+++  NEW SUPERCLASS: java.lang.Object
+	+++  NEW METHOD: PUBLIC(+) STATIC(+) io.opentelemetry.semconv.AttributeKeyTemplate<java.util.List<java.lang.Boolean>> booleanArrayKeyTemplate(java.lang.String)
+	+++  NEW METHOD: PUBLIC(+) STATIC(+) io.opentelemetry.semconv.AttributeKeyTemplate<java.lang.Boolean> booleanKeyTemplate(java.lang.String)
+	+++  NEW METHOD: PUBLIC(+) STATIC(+) io.opentelemetry.semconv.AttributeKeyTemplate<java.util.List<java.lang.Double>> doubleArrayKeyTemplate(java.lang.String)
+	+++  NEW METHOD: PUBLIC(+) STATIC(+) io.opentelemetry.semconv.AttributeKeyTemplate<java.lang.Double> doubleKeyTemplate(java.lang.String)
+	+++  NEW METHOD: PUBLIC(+) io.opentelemetry.api.common.AttributeKey<T> getAttributeKey(java.lang.String)
+	+++  NEW METHOD: PUBLIC(+) STATIC(+) io.opentelemetry.semconv.AttributeKeyTemplate<java.util.List<java.lang.Long>> longArrayKeyTemplate(java.lang.String)
+	+++  NEW METHOD: PUBLIC(+) STATIC(+) io.opentelemetry.semconv.AttributeKeyTemplate<java.lang.Long> longKeyTemplate(java.lang.String)
+	+++  NEW METHOD: PUBLIC(+) STATIC(+) io.opentelemetry.semconv.AttributeKeyTemplate<java.util.List<java.lang.String>> stringArrayKeyTemplate(java.lang.String)
+	+++  NEW METHOD: PUBLIC(+) STATIC(+) io.opentelemetry.semconv.AttributeKeyTemplate<java.lang.String> stringKeyTemplate(java.lang.String)
++++  NEW CLASS: PUBLIC(+) FINAL(+) io.opentelemetry.semconv.ClientAttributes  (not serializable)
+	+++  CLASS FILE FORMAT VERSION: 52.0 <- n.a.
+	+++  NEW SUPERCLASS: java.lang.Object
+	+++  NEW FIELD: PUBLIC(+) STATIC(+) FINAL(+) io.opentelemetry.api.common.AttributeKey<java.lang.String> CLIENT_ADDRESS
+	+++  NEW FIELD: PUBLIC(+) STATIC(+) FINAL(+) io.opentelemetry.api.common.AttributeKey<java.lang.Long> CLIENT_PORT
++++  NEW CLASS: PUBLIC(+) FINAL(+) io.opentelemetry.semconv.ErrorAttributes  (not serializable)
+	+++  CLASS FILE FORMAT VERSION: 52.0 <- n.a.
+	+++  NEW SUPERCLASS: java.lang.Object
+	+++  NEW FIELD: PUBLIC(+) STATIC(+) FINAL(+) io.opentelemetry.api.common.AttributeKey<java.lang.String> ERROR_TYPE
++++  NEW CLASS: PUBLIC(+) STATIC(+) FINAL(+) io.opentelemetry.semconv.ErrorAttributes$ErrorTypeValues  (not serializable)
+	+++  CLASS FILE FORMAT VERSION: 52.0 <- n.a.
+	+++  NEW SUPERCLASS: java.lang.Object
+	+++  NEW FIELD: PUBLIC(+) STATIC(+) FINAL(+) java.lang.String OTHER
++++  NEW CLASS: PUBLIC(+) FINAL(+) io.opentelemetry.semconv.ExceptionAttributes  (not serializable)
+	+++  CLASS FILE FORMAT VERSION: 52.0 <- n.a.
+	+++  NEW SUPERCLASS: java.lang.Object
+	+++  NEW FIELD: PUBLIC(+) STATIC(+) FINAL(+) io.opentelemetry.api.common.AttributeKey<java.lang.String> EXCEPTION_TYPE
+	+++  NEW FIELD: PUBLIC(+) STATIC(+) FINAL(+) io.opentelemetry.api.common.AttributeKey<java.lang.Boolean> EXCEPTION_ESCAPED
```

Comment:
should we proactively remove this unused (and likely to be deprecated) constant? https://github.com/open-telemetry/semantic-conventions/issues/1516

cc @lmolkova

Reply:
> should we proactively remove this unused (and likely to be deprecated) constant?

Would you imagine doing this in `semantic-conventions` or in the code generation tooling or in the code generation configuration local to `semantic-convention-java`?

Reply:
I think we would need to hack it somehow to remove it in this repo, and wait to see what happens to the attribute in semconv

not sure it's worth it, there will be other deprecated "stable" attributes in the future anyways...

Reply:
It should be really easy to exclude (a few) specific attributes by extending this list -

https://github.com/open-telemetry/semantic-conventions-java/blob/828296fad46df5651ba0d83fd8e52f09a0a14c9b/buildscripts/templates/registry/java/weaver.yaml#L3

It would felt great if we already deprecated it in semconv.

Reply:
thanks, I'll send a separate PR to try it out

Reply:
Just so we're on the same page, the principle behind removing this attribute is that we want to start with a clean slate and not stabilize anything deprecated unnecessarily. By this logic, this is the only instance where we would do this, since any attribute which is deprecated in the future will have to be maintained for backwards compatibility.

Is this your understanding?

Reply:
yes

Reply:
Makes sense to me. If for some strange reason we need this attribute, we can revert the exclusion to restore it.

Reply:
#133